client/sites-dashboard/controller.tsx

```diff
@@ -17,7 +17,7 @@ export function sitesDashboard( context: PageJSContext, next: () => void ) {
 	context.primary = (
 		<>
 			<Global styles={ globalStyles } />
-			<SitesDashboard launchStatus={ context.query.status } />
+			<SitesDashboard launchStatus={ context.query.status } siteBasePath={ context.query.return } />
```

Comment:
It's convenient, but that way, we allow to tamper with the return URL. For example, I could prepare the following URL, and if we use such an URL, we won't be able to navigate to the individual sites:

http://calypso.localhost:3000/sites-dashboard?return=%2Fme?

In the current form, we allow to navigate outside the site as well:
http://calypso.localhost:3000/sites-dashboard?return=https://example.com/

Should we handle it based on the list of allowed redirect codes (we would set them in `client/my-sites/controller.js` as a "return" param") and then perform translating the codes into actual URLs in the Sites Dashboard component? We could do this for CTA labels as well like we do for [a header in the current selector](https://github.com/Automattic/wp-calypso/blob/b36a8423eace4e272f5b01deb349ba8908cc282f/client/my-sites/sites/index.jsx#L79)

Reply:
> In the current form, we allow to navigate outside the site as well

Great catch! It's technically unsanitised user data so we really should be careful with redirecting to this query param!

We can verify that it links somewhere internal, but tbh I like your rationale for doing away with context aware links in the dashboard pdKhl6-vB-p2